azure/types.go

```diff
@@ -82,6 +82,13 @@ type TagsSpec struct {
 	Annotation string
 }
 
+// SecurityRulesSpec defines the specification for a security rule.
+type SecurityRulesSpec struct {
+	Scope        string
+	SecurityRule infrav1.SecurityRule
+	Annotation   string
```

Comment:
That's true, the only possible annotation is the security rule one. I added it as a type here in order to future-proof the type in case it needs to support more annotations, but I think that was a misunderstanding of how annotations worked like how it does for tags.

Reply:
Yeah, my sense is we won't need a mechanism like that here. And we have the tags service as an example if we do need this later.